python/tests/test_mode_decomposition.py

```diff
@@ -665,6 +668,201 @@ def _pw_amp(x):
         self.assertAlmostEqual(Tsum, Tflux, places=3)
         self.assertAlmostEqual(Rsum + Tsum, 1.00, places=2)
 
+    @parameterized.parameterized.expand(
+        [
+            (Polarization.S, 54.3, 0.0),
```

Comment:
This should really be a nonzero distance.

We got a bit "lucky" here in getting the correct phase right at the interface, as explained in https://github.com/NanoComp/meep/issues/2409#issuecomment-1462703275, but I don't want to guarantee that for future versions of Meep, and it's not something I would encourage.  (It won't work for the transmitted phase, and it will give you the wrong amplitude in any case.)

Reply:
Fixed.